tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -175,6 +175,64 @@ def definition_func(inout_field: gtscript.Field[float]):
         stmt = def_ir.computations[0].body.stmts[0]
         assert isinstance(stmt.value, gt_ir.ScalarLiteral) and stmt.value.value == 1
 
+    def test_function_import(self, id_version):
+        module = f"TestInlinedExternals_test_recursive_imports_{id_version}"
+
+        def some_function():
+            from __externals__ import const
+
+            return const
+
+        def definition_func(inout_field: gtscript.Field[float]):
+            from __externals__ import some_call
+
+            with computation(PARALLEL), interval(...):
+                inout_field = some_call()
+
+        stencil_id, def_ir = compile_definition(
+            definition_func,
+            "test_recursive_imports",
+            module,
+            externals={"some_call": some_function, "const": GLOBAL_CONSTANT},
+        )
+        assert set(def_ir.externals.keys()) == {
+            "some_call",
+            "const",
+        }
```

Comment:
For reasons that still elude me, this test is necessary. I thought this was covered below, but this uncovered the need for the additional annotate call at gtscript_frontend.py:1477.